task_templates/pipelines/python3_anomaly_detection/custom.py

```diff
@@ -1,40 +1,112 @@
+"""
+    In this example we see how to create an anomaly detection (i.e. unsupervised) pipeline
+
+    Note: in our score hook we use the same column name as regression, i.e. columns=['Predictions']
+"""
+from typing import List, Optional, Any, Dict
+import pandas as pd
+import numpy as np
 import pickle
+
 from build_pipeline import make_anomaly
 
 
 def fit(
-    X, output_dir, row_weights=None, **kwargs,
-):
-    """
-    This hook must be implemented with your fitting code, for running drum in the fit mode.
+        X: pd.DataFrame,
+        y: pd.Series,
+        output_dir: str,
+        class_order: Optional[List[str]] = None,
+        row_weights: Optional[np.ndarray] = None,
+        **kwargs,
+) -> None:
+    """ This hook MUST ALWAYS be implemented for custom tasks.
+
+    This hook defines how DataRobot will train this task.
+    DataRobot runs this hook when the task is being trained inside a blueprint.
+
+    DataRobot will pass the training data, project target, and additional parameters based on the project
+    and blueprint configuration as parameters to this function.
 
-    This hook MUST ALWAYS be implemented for custom tasks.
-    For inference models, this hook can stick around unimplemented, and won’t be triggered.
+    As an output, this hook is expected to create an artifact containing a trained object,
+    that is then used to score new data.
 
     Parameters
     ----------
-    X: pd.DataFrame - training data to perform fit on
-    output_dir: the path to write output. This is the path provided in '--output' parameter of the
-        'drum fit' command.
-    row_weights: An array of non-negative numeric values which can be used to dictate how important
+    X: pd.DataFrame
+        Training data that DataRobot passes when this task is being trained. Note that both the training data AND
+        column (feature) names are passed
+    y: pd.Series
+        Project's target column.
+    output_dir: str
+        A path to the output folder (also provided in --output paramter of 'drum fit' command)
+        The artifact [in this example - containing the trained sklearn pipeline]
+        must be saved into this folder.
+    class_order: Optional[List[str]]
+        This indicates which class DataRobot considers positive or negative. E.g. 'yes' is positive, 'no' is negative.
+        Class order will always be passed to fit by DataRobot for classification tasks,
+        and never otherwise. When models predict, they output a likelihood of one class, with a
+        value from 0 to 1. The likelihood of the other class is 1 - this likelihood. Class order
+        dictates that the first element in the list will be the 0 class, and the second will be the
```

Comment:
theoretically, users could set 0 as positive class in DataRobot. so a more universally accurate statement would be "first element in the class_order list is the name if the class considered negative inside DR's project, and the second is the name of the class that is considered positive."

Reply:
Agreed